kartothek/io_components/metapartition.py

```diff
@@ -686,7 +686,17 @@ def load_dataframes(
                 and self.partition_keys
                 and set(table_columns) == set(self.partition_keys)
             ):
-                table_columns_to_io = None
+                # find a single dummy column that we can load
+                if table_columns is not None:
+                    table_columns_to_io_candidates = sorted(
+                        set(self.table_meta[table].names) - set(table_columns)
+                    )
+                    if table_columns_to_io_candidates:
+                        table_columns_to_io = [table_columns_to_io_candidates[0]]
+                    else:
+                        table_columns_to_io = None
+                else:
+                    table_columns_to_io = None
```

Comment:
What's the background for this change? Can you add a comment in the code? I cannot see any details in the commit message

Reply:
"only load a single dummy column in MetaPartition.load_dataframes" is the commit message. This case happens if the user does not request any columns but we still need at least a single dummy column to reconstruct a non-broken DF. W/o this change, we just loaded ALL available columns which is way too much.

Reply:
Ok let me rephrase it: I suspected the "else we load everything" answer already but I'm not sure if this is clear in the code once a bit of time passes. Also, I'm pretty sure this was possible at some point. Do you know if anything in arrow changed in that regard?

Reply:
I took a brief look at the serialization module and it looks like if we pass `colums=[]` we get back a dataframe with an index the size of the number of rows of the Parquet file. Wouldn't that be faster? (Or is that not suitable for index reconstruction afterwards?)

Reply:
Let me explain it based on @lr4d question:

if you pass an empty column list, you want the set of rows (at least the DF len) to be restored. The issue is that not all serializers have dealed with this edge case correctly in the past. It was implemented here because it is the only location where the reader knows about potential dummy columns. That said, I removed the special case handling all together because to my knownledge, all serializers can now deal with this correctly.

Reply:
Thanks for the explanation @marco-neumann-jdas